### DISCUSSION TITLE
Altitude seems problematic when using INS/GNSS fusion

### DISCUSSION THREAD

**Discussion by clementbonnel** [rodralez/NaveGo#67] [Q&A] [answered]
Dear team of NaveGo, 

I am looking to use both ins and gnss data to obtain a more precise position of a vehicle. 
When using ins_gnss function I have found myself with an altitude error very high. 
Could you tell me, what could be a cause of such a difference and how can I improve it ? 
I have tried to change std, or imu.ini__align but it doesn't work. 

Thank you. 

Clément Bonnel.  
 
![latlonalterror](https://user-images.githubusercontent.com/89390957/130590837-7d0b378c-f75d-4d5a-9013-71fe78b6a923.png)

**Comment by BrunoOtavio98** [comment]
Hello @clementbonnel, can you post your dataset here? It will help to find where is the problem.

**Reply by BrunoOtavio98** [reply]
Also your error model for both gps and imu.

**Reply by clementbonnel** [reply]
Hello @BrunoOtavio98 thank you for your quick reply. 
Here is a link to the files. https://drive.google.com/drive/folders/1QDI-3IoqUl3Px94Vp3dNTSid2gkZ3V3a?usp=sharing

**Reply by clementbonnel** [reply]
There are four files in what I have sent you. 
The ADIS file is the ins file of the vehicle, the ADISsta file is the ins file of the vehicle in a static position to determine the allan variance,
the ASTERX file is  the gnss file, and the ASTERX1 is the file used for the std. 
For the ins errors I have used allan_imu.m with ADISsta and added imu.gb_sta=[1.2e-6,1e-6,0];
imu.ab_sta=[14e-6,20e-6,10e-6];imu.ini_align=[1,1,311.17]*0.0175;
imu.ini_align_err=[0.0873,0.0873,0.0873];
Finally for the gnss errors gnss.stdm=[2.9167,1.8912,-5.6756];gnss.stdv=[0.000025,0.0001,-0.0005];
gnss.larm=[-1.34,-1.135,1.665]';
gnss.freq=1;
gnss.zupt_th=0.3;
gnss.zupt_win=1;
gnss.eps=0.1;
Please tell me wether I am clear or not. 
Again thank you for helping me

**Comment by rodralez** [comment, marked as answer]
Hello Clément,

Generally, you have to tune the IMU error profile in order the Kalman filter finds better estimation of the states vector. Besides, check that the Z acceleration (imu.fb( : ,  3)) are negative values around -9.8 m/s^2.

Best regards.

**Reply by clementbonnel** [reply]
Hello Rodrigo, thank you for your reply. 

I am trying to understand , what do you mean by tuning the IMU error profile ? 
Do you mean that I need to find a better IMU error profile ? 
If so, is there such a program ?  or is it better if I try running with different values ? 
For the Z acceleration I had already checked it, thank you. 

Best regards.

**Reply by BrunoOtavio98** [reply]
Hello @clementbonnel , sometimes  you cant just trust the allan variance output for error profile, mainly when your accelerometer or gyroscope dont follow a normal distribution. So, what rodralez wants to say is that, after you get the error profile with allan variance, you have to hand tunning the values and see whats the output from kalman filter. You have to repeat that untill you get a better output. For me, i took two weeks untill i got something good.

**Reply by rodralez** [reply]
@BrunoOtavio98 , you read my mind! :-)

@clementbonnel , bear in mind that the Allan variance is a method that was developed to investigate the stability of crystal oscillators and atomic clocks, so I am not sure that is the best tool to infer a noisy IMU error profile.

Best regards.

**Reply by clementbonnel** [reply]
Hello again, 

Thank you both for your replies they are very helpful. 

I am looking at the example ekinox to have a working example of the allan imu. 
In the example, the IMU error profile is already determined, and I would like to use allan_imu to see if the results are the same.  
Do you still have the static data used to obtain the allan variance ? If so, could you show it to me ? 

Thank you again for all your help.

**Reply by clementbonnel** [reply]
Furthermore i have run the example and found these errors, are these errors expected ? 
![errorekinoxlatlonh](https://user-images.githubusercontent.com/89390957/131513256-9786e653-6203-4a7e-9e91-26ae6794d779.png)

**Reply by clementbonnel** [reply]
*ekinox example

**Reply by rodralez** [reply]
Hi, you can download the Ekinox static dataset from the following link (620 MB):

https://nube.ingenieria.uncuyo.edu.ar/s/EMAGGQpT2H2j3LD

Regards.
